streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -243,11 +253,25 @@ private void startRestoration(final Set<TopicPartition> initialized,
 
     private void logRestoreOffsets(final TopicPartition partition,
                                    final long startingOffset,
+                                   final long offsetLimit,
                                    final Long endOffset) {
-        log.debug("Restoring partition {} from offset {} to endOffset {}",
+        // Note that we will restore to the smaller of offsetLimit or endOffset, which is how
+        // StateRestorer.hasCompleted works. It would be nice to not have this knowledge in
+        // multiple places.
+        //
+        // endOffset is boxed here. Is it safe to assume it is not null? It seems other code, such
+        // as StateRestorer.hasCompleted assume this.
+        final long lastOffset;
```

Comment:
According to the comment on line 158, the consumer API does not guarantee that `endOffset` is not `null`. The javadoc of `KafkaConsumer#endOffsets()` say that there will be a non-`null` `endOffset` for each given partition (at least in my interpretation). @guozhangwang, WDYT?
 
Please remove comment once this issue is solved.